File: CoWorkHub/CoWorkHub.Model/SpaceUnit.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class SpaceUnit
    {
        public int SpaceUnitId { get; set; }
        public int WorkingSpaceId { get; set; }
        public string Name { get; set; } = null!;
        public string Description { get; set; } = null!;
        public int WorkspaceTypeId { get; set; }
        public int Capacity { get; set; }
        public decimal PricePerDay { get; set; }
        public string StateMachine { get; set; } = null!;
        public bool IsDeleted { get; set; } = false;
        public virtual WorkingSpace WorkingSpace { get; set; } = null!;
        public virtual ICollection<SpaceUnitResources> SpaceUnitResource { get; set; } = new List<SpaceUnitResources>();
        public virtual WorkspaceType WorkspaceType { get; set; } = null!;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/space_unit.dart
import 'package:json_annotation/json_annotation.dart';
import 'workspace_type.dart';
import 'space_unit_resources.dart';
import 'working_space.dart';
import 'space_unit_image.dart';

part 'space_unit.g.dart';

@JsonSerializable(explicitToJson: true)
class SpaceUnit {
  int spaceUnitId;
  int workingSpaceId;
  String name;
  String description;
  int workspaceTypeId;
  int capacity;
  double pricePerDay;
  String stateMachine;
  bool isDeleted;
  WorkingSpace? workingSpace;
  List<SpaceUnitResources> spaceUnitResources;
  WorkspaceType? workspaceType;
  List<SpaceUnitImage> spaceUnitImages;

  SpaceUnit({
    required this.spaceUnitId,
    required this.workingSpaceId,
    required this.name,
    required this.description,
    required this.workspaceTypeId,
    required this.capacity,
    required this.pricePerDay,
    required this.stateMachine,
    this.isDeleted = false,
    this.workingSpace,
    required this.spaceUnitResources,
    this.workspaceType,
    required this.spaceUnitImages,
  });

  factory SpaceUnit.fromJson(Map<String, dynamic> json) =>
      _$SpaceUnitFromJson(json);

  Map<String, dynamic> toJson() => _$SpaceUnitToJson(this);
}
